pkg/apis/authentication/v1alpha1/policy.go

```diff
@@ -51,8 +56,34 @@ type PolicySpec struct {
 	istiov1alpha1.Policy
 }
 
+// Taken from https://github.com/michaelkipper/istio-client-go/commit/7c8e95b5d9220d47c107bb6f3b0b71fbc8af3ef7#diff-1c6fa6bfc320013a249f4b6d0ccdd928R65
+func (p *PolicySpec) MarshalJSON() ([]byte, error) {
+	buffer := bytes.Buffer{}
+	writer := bufio.NewWriter(&buffer)
+	marshaler := jsonpb.Marshaler{}
+	err := marshaler.Marshal(writer, &p.Policy)
+	if err != nil {
+		log.Printf("Could not marshal PolicySpec. Error: %v", err)
+		return nil, err
+	}
+
+	writer.Flush()
+	return buffer.Bytes(), nil
+}
+
+func (p *PolicySpec) UnmarshalJSON(b []byte) error {
+	reader := bytes.NewReader(b)
+	unmarshaler := jsonpb.Unmarshaler{}
+	err := unmarshaler.Unmarshal(reader, &p.Policy)
+	if err != nil {
+		log.Printf("Could not unmarshal PolicySpec. Error: %v", err)
+		return err
+	}
+	return nil
+}
+
 // DeepCopyInto is a deepcopy function, copying the receiver, writing into out. in must be non-nil.
 // Based of https://github.com/istio/istio/blob/release-0.8/pilot/pkg/config/kube/crd/types.go#L450
-func (in *PolicySpec) DeepCopyInto(out *PolicySpec) {
-	*out = *in
+func (p *PolicySpec) DeepCopyInto(out *PolicySpec) {
```

Comment:
*in and *out is a generic name for DeepCopy functions, I would suggest keeping it the same.

Reply:
Didn't think I changed that... will revert to original.